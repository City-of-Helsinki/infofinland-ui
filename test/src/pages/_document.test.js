import { render } from '@testing-library/react';
import MyDocument from '@/pages/_document';
import React from 'react';

jest.mock('next/document', () => ({
  __esModule: true,
  ...jest.requireActual('next/document'),
  Html: jest.fn().mockImplementation((props) => (
    <div data-dir={props.dir} data-lang={props.lang} />
  ))
}));

describe('pages', () => {
  describe('_document', () => {
    it('should render', async () => {
      const { container } = render(<MyDocument locale="fi" />)
      expect(container).toMatchSnapshot()
    })

    it('should render with rtl locale', async () => {
      const { container } = render(<MyDocument locale="ar" />)
      expect(container).toMatchSnapshot()
    })
  })
})
