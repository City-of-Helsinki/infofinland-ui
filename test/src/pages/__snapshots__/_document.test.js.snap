// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`pages _document should render 1`] = `
<div>
  <div
    data-dir="ltr"
    data-lang="fi"
  />
</div>
`;

exports[`pages _document should render with rtl locale 1`] = `
<div>
  <div
    data-dir="rtl"
    data-lang="ar"
  />
</div>
`;
